src/rules/selector-no-id/index.js

```diff
@@ -19,6 +19,7 @@ export default function (actual) {
     root.walkRules(rule => {
       selectorParser(selectorAST => {
         selectorAST.eachId(id => {
+          if (/#{.+}/.test(id)) { return }
```

Comment:
I'd be willing to work in this special case. @jeddy3 what do you think? I guess the only time it would backfire is if somebody wanted to use a real id that like `{something}`.